AstSemantics.md

```diff
@@ -111,6 +112,9 @@ Integer loads can specify a *storage size* which is smaller than the result type
 well as a signedness which determines whether the bytes are sign- or zero-
 extended into the result type.
 
+Each `load` and `store` specifies the [linear memory index](Modules.md#index-spaces)
+to operate on with an unsigned integer immediate.
+
```

Comment:
> From a C++ pov, multiple memories would correspond to the (also statically numbered) address space in LLVM (and address spaces would provide a direct way to access this functionality from C++).

Looking for what that feature is used for in LLVM, the best I can find is (1) thread-local storage on x86 and (2) gc things. However, both of those would be done very differently in wasm, wouldn't they?

Are there other uses of this feature in LLVM that would be relevant for us?